src/main/java/net/neoforged/neoform/runtime/actions/ExtensibleClasspath.java

```diff
@@ -33,12 +33,25 @@ public List<ClasspathItem> getEffectiveClasspath() {
 
     public void addMinecraftLibraries(Collection<MinecraftLibrary> libraries) {
         for (var library : libraries) {
-            if (library.rulesMatch() && library.getArtifactDownload() != null) {
+            if (library.rulesMatch() && library.getArtifactDownload() != null && !containsLibrary(library)) {
                 add(ClasspathItem.of(library));
             }
         }
     }
 
+    private boolean containsLibrary(MinecraftLibrary library) {
+        var mavenCoordinate = library.getMavenCoordinate();
+
+        for (var item : additionalClasspath) {
+            if (item instanceof ClasspathItem.MinecraftLibraryItem(MinecraftLibrary otherLib) && otherLib.equals(library)) {
+                return true;
+            } else if (item instanceof ClasspathItem.MavenCoordinateItem mavenCoordinateItem && mavenCoordinateItem.coordinate().equals(mavenCoordinate)) {
```

Comment:
Are you sure that this will work? It seems to me that `MavenCoordinate#equals` will also compare the versions.